### DISCUSSION TITLE
DataGrid Control cannot be resolved in Release configuration

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#12091] [Q&A]
The upgrade from 0.10.19 to 11.0.0 of my project has been finished.

But I encounter a strange behavior since compilation fails in Release configuration while in Debug configuration everything runs fine.

I am using Rider 2023.1 and I am not sure if the issue is from Avalonia or Rider.

In Debug configuration the package Avalonia.Controls.DataGrid is listed in Rider as implicetely installed and this is not in Release configuration. Here the packages is missing.

Is this a Bug?

**Comment by jp2masa** [comment]
https://github.com/AvaloniaUI/Avalonia/discussions/8463#discussioncomment-3101982

**Comment by drei97** [comment]
My point is not to solve the issue on my side. To add a package in .csproj by hand or by nuget is no rocket science.

I am wondering why there are differences in the implicitly settings in the nuget packages for Release and for Debug.

Is this by intention and if so, why? I never recognized such behavior on any packages from other publisher. To be honest, I do not have any idea about any insides of packages building. I am just confused.

**Reply by timunie** [reply]
In debug mode you can use devtools to inspect your xaml via hitting F12. This package has a reference to DataGrid, so its pulled in by DevTools. But in release mode, this package is not installed so its dependencies are not. Check your csproj file for reference. 

The solution is not ideal, but still very useful tool, so it is what it is. 

I hope it's clearer now.
